### PR TITLE
Added Egg Drop

### LINKED ISSUE #994: Egg Drop Problem in C and C++

### PR DESCRIPTION
**Pull request by somya-kapoor**
Fixes #994 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

-  Added Egg Drop problem in C and C++

#### Languages Used:
- C
- C++

#### Files Added:
- Egg_Drop.c
- Egg_Drop.cpp

### CODE REVIEW

File: Egg_Drop/Egg_Drop.c [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# include <stdio.h>
+
+// To get maximum of two integers 
+#define max( a, b) ((a > b)? a : b)
```
Comment (marwahmanbir) [inline Egg_Drop/Egg_Drop.c]: Hey! your code looks good. Just a few changes.
Remove space before ``a``.

File: Egg_Drop/Egg_Drop.c [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            else if(e == 1)
+            {
+                eggfloor[e][f] = f;
+            }
+            else
+            {
+                // Fill rest of the entries in table using optimal substructure 
+                // property 
+                int min = 9999;
+                for (int i = 1; i <= f ; i++)
```
Comment (marwahmanbir) [inline Egg_Drop/Egg_Drop.c]: remove space after ``i <= f``

File: Egg_Drop/Egg_Drop.c [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    }
+    return eggfloor[eggs][floors];
+}
+
+int main() 
+{ 
+    int eggs = 2, floors = 10; 
+    printf ( "The number of eggs: %d\n" , eggs ) ;    
+    printf ( "The number of floors: %d\n" , floors ) ;
+    printf ( "Minimum number of trials:  %d " , eggDrop(eggs, floors));
```
Comment (marwahmanbir) [inline Egg_Drop/Egg_Drop.c]: In ``printf`` statements spaces are not required around opening and closing parenthesis as far as I know. Please confirm this and make the changes if required.

Comment (marwahmanbir) [reply Egg_Drop/Egg_Drop.c]: Also, remove spaces before commas.

File: Egg_Drop/Egg_Drop.cpp [hunk 4]
```diff
@@ -0,0 +1,59 @@
+# include <stdio.h>
+# include <iostream>
+using namespace std; 
+
+// To get maximum of two integers 
+#define max( a, b) ((a > b)? a : b)
```
Comment (marwahmanbir) [inline Egg_Drop/Egg_Drop.cpp]: Fix spacing.

File: Egg_Drop/Egg_Drop.cpp [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+int eggDrop(int eggs, int floors)
+{
+    /* A 2D table where entery eggFloor[i][j] will represent minimum 
+       number of trials needed for e eggs and f floors. */
+    int eggfloor[eggs][floors]; 
+    for (int e = 1; e <= eggs ; e++)
```
Comment (marwahmanbir) [inline Egg_Drop/Egg_Drop.cpp]: Fix spacing.

File: Egg_Drop/Egg_Drop.cpp [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        for (int f = 0; f <= floors ; f++)
```
Comment (marwahmanbir) [inline Egg_Drop/Egg_Drop.cpp]: Fix spacing.

File: Egg_Drop/Egg_Drop.cpp [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+            else if(e == 1)
+            {
+                eggfloor[e][f] = f;
+            }
+            else
+            {
+                // Fill rest of the entries in table using optimal substructure 
+                // property 
+                int min = INT_MAX;
+                for (int i = 1; i <= f ; i++)
```
Comment (marwahmanbir) [inline Egg_Drop/Egg_Drop.cpp]: Fix spacing.

File: Egg_Drop/Egg_Drop.cpp [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+                eggfloor[e][f] = min;
+            }
+        }
+    }
+    return eggfloor[eggs][floors];
+}
+
+int main() 
+{ 
+    int eggs , floors;
```
Comment (marwahmanbir) [inline Egg_Drop/Egg_Drop.cpp]: Fix spacing.

**Comment by somya-kapoor** [comment]
@marwahmanbir I have done the required changes. Please review

**Pull request merged**